### PR TITLE
Separate BTT Manta Environments

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

Use seperate / Manta-specific environments as to not break Ender-3 V2 LCD support in the existing STM32G0-based environment.

### Related Issues

- https://github.com/MarlinFirmware/Marlin/pull/25001#issuecomment-1442833370

### CODE REVIEW

**Comment by thisiskeithb** [comment]
I still need to verify if the TFT port works on the Manta boards, but at least this won't break the Ender-3 V2 LCD support 🤷

File: buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/PeripheralPins.c [hunk 1]
```diff
@@ -253,7 +252,6 @@ WEAK const PinMap PinMap_UART_RX[] = {
   {PC_11_ALT1, USART4,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_USART4)},
   {PD_2,       USART5,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF3_USART5)},
   {PD_6,       USART2,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF0_USART2)},
-  {PD_9,       USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF4_USART3)},
```
Comment (thinkyhead) [inline buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/PeripheralPins.c]: In the Arduino framework the `PinMap_UART_RX` table is used by the `HardwareSerial` constructor and the `digitalPinHasSerial` macro. And the STM32 framework uses it for `uart_init`. But it's the `HAVE_HWSERIAL#` macros (and/or `SERIAL_UART_INSTANCE`) that determine whether a port will be functional. So, should this line be included even though the serial port is not being activated elsewhere?

Comment (thisiskeithb) [reply buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/PeripheralPins.c]: @bigtreetech would have to answer that.

**Comment by mriscoc** [comment]
Thanks

**Comment by thisiskeithb** [comment]
> Thanks

@mriscoc Does that mean this PR fixes your issue? I should be able to test BTT's TFTs on Manta boards tomorrow.

**Comment by mriscoc** [comment]
> this PR fixes your issue?

I don't have that board to do testing, I did receive reports from users and started to do some cherry picks until I got the offending PR.

So, I will be able to do the test only if I implement this fix, compile a version and send it to a collaborator with that board and screen.

**Comment by thisiskeithb** [comment]
> So, I will be able to do the test only if I implement this fix, compile a version and send it to a collaborator with that board and screen.

Please do so we can merge this without needing more follow-up.

**Comment by thisiskeithb** [comment]
Below is the status of cherry picking this PR for testing with a BTT TFT (running under "Touch Mode" / over serial using the board's default TFT pins) with the following STM32G0-based boards:

- [x] BTT SKR Mini E3 V3
- [x] BTT Manta E3 EZ
- [ ] BTT Manta M4P
- ~~BTT Manta M5P~~ No TFT port
- ~~BTT Manta M8P V1.0~~ No TFT port
- ~~BTT Manta M8P V1.1~~ No TFT port

**Comment by thisiskeithb** [comment]
Other than the change that thinkyhead reverted in https://github.com/MarlinFirmware/Marlin/pull/25431/commits/d2b2696a6972b72d39c11970eb365196d05fa856, this should work with the Ender-3 V2 LCD again (or at least my original Manta boards PR wasn't the source of the problem).

**Comment by thisiskeithb** [comment]
Actually, with the reverting change in https://github.com/MarlinFirmware/Marlin/pull/25431/commits/d2b2696a6972b72d39c11970eb365196d05fa856, what's the point of breaking out these boards to a separate environment?

It would seem the only real fix needed was reverting the changes in [variant_MARLIN_STM32G0B1RE.h](https://github.com/MarlinFirmware/Marlin/pull/25431/commits/23c3d2ed1d69f9ffe10be909ab9d50896e4f1ab8#diff-7f424d14ece4a098e7b5cd1119e300cdda5b7c2dbb3a52a47599d8c34efef4e4).

**Comment by thisiskeithb** [comment]
Reverted https://github.com/MarlinFirmware/Marlin/pull/25431/commits/d2b2696a6972b72d39c11970eb365196d05fa856 since that is already handled in ini / STM32G0B1RE_manta_btt env flags.

**Comment by thisiskeithb** [comment]
The changes in https://github.com/MarlinFirmware/Marlin/pull/25431/commits/d2b2696a6972b72d39c11970eb365196d05fa856 are required since just having them in the .ini/`STM32G0B1RE_manta_btt` env is not enough.

I also tried dropping just the `-DPIN_SERIAL3_RX=PD_9 -DPIN_SERIAL3_TX=PD_8` flags from the `STM32G0B1RE_manta_btt` env (and leaving https://github.com/MarlinFirmware/Marlin/pull/25431/commits/d2b2696a6972b72d39c11970eb365196d05fa856 intact), but that doesn't work. You must have both.

**Comment by thinkyhead** [comment]
> I also tried dropping just the `-DPIN_SERIAL3_RX=PD_9 -DPIN_SERIAL3_TX=PD_8` flags from the `STM32G0B1RE_manta_btt` env (and leaving d2b2696 intact), but that doesn't work. You must have both.

My guess is the only thing that really needs to be reverted is the line that sets `ENABLE_HWSERIAL3` in the variant, moving it to the env. Everything else is harmless. For example, we could leave the defines for `PIN_SERIAL3_RX`/`TX` in the variant if we wanted to, even if they're not used. The platform ends up picking them up either way.

**Comment by ebraiman** [comment]
@thisiskeithb Are there Manta Configurations you want tested?  I have everything except Manta M5P.  Also have CB1, CM4 and CM4 with 8/16GB EMMc.  I'm aware 16GB has issues.

**Comment by ebraiman** [comment]
@thinkyhead Tested TFT with Marlin Bugfix from today.  TFT display does not connected at 250000 on Manta E3EZ.  TFT version is TFT35_E3_v3.0 FW V27.x Apr 9 2022.  Will upgrade TFT FW and check again.

**Comment by thisiskeithb** [comment]
> TFT display does not connected at 250000 on Manta E3EZ.

115200 worked while adding support for this board and I just confirmed that it still does.

250000 also works fine.

---

This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- [Marlin Documentation](https://marlinfw.org) hosted on MarlinFW.org
- [Marlin Forum](https://forums.reprap.org/list.php?415) hosted on RepRap.org
- [Marlin Firmware](https://www.facebook.com/groups/1049718498464482/) Facebook Group 
- [Marlin Firmware for 3D Printers](https://www.facebook.com/groups/3Dtechtalk/) Facebook Group
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- [Marlin Firmware](https://discord.gg/n5NJ59y) Discord server

After seeking help from the community, if the consensus points to a bug in Marlin, then you should post a [bug report](https://github.com/MarlinFirmware/Marlin/issues/new/choose).

**Pull request merged**